lib/buffer-search.coffee

```diff
@@ -58,6 +59,17 @@ class BufferSearch
       ResultsMarkerLayersByEditor.set(editor, layer)
     layer
 
+  checkSanity: (findPattern) ->
+    findOptions = new FindOptions(@findOptions.serialize())
+    findOptions.set({findPattern})
+    if regex = findOptions.getFindPatternRegex()
+      try
+        buffer = new TextBuffer('')
+        sane = true
+        buffer.scan regex, ({range}) -> sane = false
+        return sane
+    false
```

Comment:
I think this could be simplified to

```coffee
if regex = findOptions.getFindPatternRegex()
  not regex.test('')
else
  false
```

since the `TextBuffer.scan` method just uses the normal JS regex APIs.